src/main/scala/pl/multiplex/services/impl/ConsumerServiceImpl.scala

```diff
@@ -0,0 +1,33 @@
+package pl.multiplex.services.impl
+
+import org.springframework.stereotype.Service
+import pl.multiplex.dao.ConsumerDao
+import pl.multiplex.exceptions.InvalidFirstNameOrSecondNameException
+import pl.multiplex.models.Consumer
+import pl.multiplex.services.ConsumerService
+
+
+@Service
+class ConsumerServiceImpl(val consumerDao: ConsumerDao) extends ConsumerService {
+
+  @throws(classOf[InvalidFirstNameOrSecondNameException])
```

Comment:
co nam daje ten throws?

Reply:
W Javie w takim przypadku mógłby paść wyjątek Chcecked i obsługa bądź przekazanie do klasy wywołującej byłoby obowiązkowe.W Scali o ile dobrze rozumiem wszystkie wyjątki są Unchecked, więc bez @throws też by to działało. Jednak adnotacja oznacza metodę dla programisty pracującego z nią oraz dla kompilatora. 